app/src/main/java/org/schabi/newpipe/player/BasePlayer.java

```diff
@@ -1285,4 +1301,44 @@ private boolean isPlaybackResumeEnabled() {
         return prefs.getBoolean(context.getString(R.string.enable_watch_history_key), true)
                 && prefs.getBoolean(context.getString(R.string.enable_playback_resume_key), true);
     }
+
+    public void setTimer(int hourOfDay, int minute, Context ctx) {
+        long time = ((hourOfDay * 60) + minute) * 60 * 1000;
```

Comment:
I think this is a logical error here.
I suggested you to use the number picker which would indeed return the remaining hours and minutes to pause the player. But when using the TimePicker you get the actual time of the day wich means the player will be paused at a completely different time. You add the selected time to the current time. Selecting 11:45AM means that the timer will pause the player in 11 hours and 45 minutes, but not at 11:45AM.But the TimePicker is intended to give people a specific time to select.
For this reason, the best thing would be to implement something which lets you pick the remaining time.